Fix mount path

tino1b2be opened the pull request:
Added code to account for the change in mount path. If an unmount fails, docker-xylem will attempt to unmount using old mount paths.

tino1b2be committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into fix-mount-path

tino1b2be committed:
Added code to replace utils.fork instead of mount and unmount.

tino1b2be committed:
Fix: accounted for the change in GlusterFS error codes.
Instead of searching for the phrase, search for the keywords instead to check if the volume is not mounted.

tino1b2be committed:
Fix: accounted for the change in GlusterFS error codes.
Instead of searching for the phrase, search for the keywords instead to check if the volume is not mounted.

tino1b2be committed:
Added a new test for /Volume.Unmount to test the mount path change

tino1b2be committed:
Created a function to create a list of old paths.
Created this function after the previous code threw exceptions when trying to create a list of old paths.

tino1b2be committed:
Fixed PEP-8 errors.

tino1b2be committed:
Changed the test mount path.
Changed the test mount path because Travis CI could not use /mnt (no permissions)

tino1b2be committed:
Changed the mount path inside the docker-xylem yml config file.
Also removed an unnecessary "/"

jerith commented on docker_xylem/service.py:
```diff
@@ -28,8 +28,14 @@ def __init__(self, config):
+        )
+        self.old_paths = {
+            '/var/lib/docker/volumes',
+            'var/lib/docker-xylem',
```
I don't think we need `var/lib/docker-xylem`. Also, `old_paths` should probably be configurable and not default to having `/var/lib/docker/volumes` in it.

jerith commented on docker_xylem/service.py:
```diff
@@ -102,12 +115,30 @@ def unmount_volume(self, request, data):
+        paths = []
+        for path in self.old_paths:
+            paths.append(os.path.join(path, name))
+        return paths
```
This could be replaced by a list comprehension:
`return [os.path.join(path, name) for path in self.old_paths]`

jerith commented on docker_xylem/tests/test_docker_xylem.py:
```diff
@@ -117,3 +126,43 @@ def test_capabilities(self):
+    def test_unmount(self):
+        """
+        Test for /Volume.Unmount
+        """
```
Can we break this up into multiple tests?

jerith reviewed (commented):
Looks good so far.

tino1b2be committed:
Shortened the get_paths function.

tino1b2be committed:
Changed the program to load old mount path from config file.
Added old mount paths in the config file.
Updated the tests to use a config with old mont paths

tino1b2be committed:
Created two separate tests fr unmounting.
one test tries to unmount if the mount path has not been changed. The other test tries to unmount if the mount path has been changed.

tino1b2be committed:
Fixed PEP-8 errors

tino1b2be committed:
Added an assertion to check if old mont paths were loaded correctly.
Corrected an error in the test config.

tino1b2be committed:
Added an assertion to check if old mont paths were loaded correctly.
Corrected an error in the test config.

jerith commented on docker_xylem/service.py:
```diff
@@ -102,12 +112,28 @@ def unmount_volume(self, request, data):
+                for path in paths:
+                    yield self._umount_fs(path)
+
+                defer.returnValue({"Err": None})
```
Would it make sense to always try unmount from old paths as well? What if we somehow ended up with the same volume mounted in multiple places?

tino1b2be committed:
Changed te code to unmount from all possible paths.

jerith reviewed (approved):
:+1:
